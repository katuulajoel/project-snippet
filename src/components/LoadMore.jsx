import PropTypes from "prop-types";
import React from "react";
import Button from "./Button";
import Progress from "./Progress";

const LoadMore = ({
  children,
  onLoadMore,
  hasMore = true,
  isLoadingMore = false,
  variant,
  size = "sm",
  type = "button",
}) => {
  return isLoadingMore ? (
    <Progress />
  ) : hasMore ? (
    <div className={`${type == "button" ? "text-center" : "text-link"}`}>
      <Button
        {...(variant ? { variant } : {})}
        {...(size ? { size } : {})}
        size="sm"
        onClick={onLoadMore}
      >
        {children || "Load More"}
      </Button>
    </div>
  ) : null;
};

LoadMore.propTypes = {
  children: PropTypes.string,
  hasMore: PropTypes.bool,
  isLoadingMore: PropTypes.bool,
  onLoadMore: PropTypes.any,
  size: PropTypes.string,
  type: PropTypes.string,
  variant: PropTypes.any,
};

export default LoadMore;
