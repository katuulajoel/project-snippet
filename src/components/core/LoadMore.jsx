/* eslint-disable react/prop-types */
import React from "react";
import Button from "./Button";
import Progress from "./Progress";

const LoadMore = ({
  children,
  onLoadMore,
  hasMore = true,
  isLoadingMore = false,
  variant,
  size = "sm",
  type = "button",
}) => {
  return isLoadingMore ? (
    <Progress />
  ) : hasMore ? (
    <div className={`${type == "button" ? "text-center" : "text-link"}`}>
      <Button
        {...(variant ? { variant } : {})}
        {...(size ? { size } : {})}
        size="sm"
        onClick={onLoadMore}
      >
        {children || "Load More"}
      </Button>
    </div>
  ) : null;
};

export default LoadMore;
