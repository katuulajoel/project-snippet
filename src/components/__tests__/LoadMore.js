import React from "react";
import renderer from "react-test-renderer";
import LoadMore from "../LoadMore";

describe("Auth layout test", () => {
  it("Snapshot test for Login component", () => {
    const tree = renderer.create(<LoadMore />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
