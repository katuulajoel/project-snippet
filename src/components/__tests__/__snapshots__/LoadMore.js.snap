// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth layout test Snapshot test for Login component 1`] = `
<div
  className="text-center"
>
  <button
    className="btn btn-default   btn-sm"
    form={null}
    type="button"
  >
    Load More
  </button>
</div>
`;
